src/gurt/telemetry.c

```diff
@@ -281,54 +309,29 @@ d_tm_init(int id, uint64_t mem_size)
 }
 
 /**
- * Initialize an instance of the telemetry and metrics API for a client
- * application process.  By design, this initialization function enables
- * read/write serialization which provides client thread safety when accessing
- * the same metrics asynchronously.
- *
- * \param[in]	client_id	Identifies the client process amongst others on
- *				the same machine
- * \param[in]	mem_size	Size in bytes of the shared memory segment that
- *				is allocated
- *
- * \return		D_TM_SUCCESS		Success
- *			-DER_NO_SHMEM		Out of shared memory
- *			-DER_EXCEEDS_PATH_LEN	Root node name exceeds path len
+ * Releases resources claimed by init
  */
-int
-d_tm_init_client(int client_id, uint64_t mem_size)
+void d_tm_fini(void)
 {
 	int	rc = 0;
 
-	d_tm_serialization_enabled = true;
-	rc = d_tm_init(client_id, mem_size);
-	return rc;
-}
+	if (d_tm_shmem_root == NULL)
+		return;
 
-/**
- * Releases resources claimed by init
- * Currently, only detaches from the shared memory to keep the memory and the
- * shared mutex objects available for the client to read data even when the
- * producer has gone offline.
- */
-void d_tm_fini(void)
-{
-	if (d_tm_shmem_root != NULL) {
-		/**
-		 * If we decide to free the mutex objects on shutdown for the
-		 * nodes that have them, call:
-		 * d_tm_free_node(d_tm_shmem_root, d_tm_root);
-		 *
-		 * Destroying the mutex objects on shutdown makes them
-		 * unavailable to a client process that may want to read the
-		 * data.  To remedy that, can implement a reference count that
-		 * monitors the number of users of the shared memory segment
-		 * and only frees the resources if it's the last one using it.
-		 */
-		shmdt(d_tm_shmem_root);
-		d_tm_shmem_root = NULL;
-		d_tm_root = NULL;
+	shmdt(d_tm_shmem_root);
+	if (!d_tm_retain_shmem) {
+		rc = shmctl(d_tm_shmid, IPC_RMID, NULL);
+		if (rc < 0)
+			D_ERROR("Unable to remove shared memory segment. "
+				"rc = %d\n", rc);
```

Comment:
print error string?

Reply:
The other errors in telemetry.c are printing an integer rc.  If we want to convert those to strings, I can make another PR to address that.

Reply:
Addressed all of them in the latest update.